### ISSUE TITLE
Starling's new AssetManager

### ISSUE THREAD

**Issue by ValentinRadoi** [openfl/starling#88]
Hello!

Starling has recently added a redesigned AssetManager to its master branch, offering much greater flexibility in how assets are dealt with (https://forum.starling-framework.org/topic/beta-new-assetmanager). This can work in conjunction with a new extension for caching assets transparently on disk, for mobile apps (http://wiki.starling-framework.org/extensions/asset-caching).

I know this is very recent stuff (only 5 days ago merged to master), but I was wondering if you have plans to port this to the Haxe Starling any time soon or it's not such a priority at this moment. I wanted to know how do you plan to approach this:
- wait for Starling 2.4 to be fully ready and released, then port all of its improvements together, including this new AssetManager or
- try to extract only the new AssetManager (I think it's pretty much confined to the /assets package, with just a minor change in the Texture class) earlier and paste it on top of the current Haxe Starling 2.2.1 that we already have.

I'm asking because I'm about to start developing a new game with Haxe Starling (my first Haxe experience, after coming from an AIR/Starling background + a year spent with Unity), and one of the foundation matters I wanted to address as early as possible was the way assets will be bundled, downloaded, cached and updated across all platforms I'm interested in (html5, Android and iOS). So obviously this new AssetsManager is a key point in my plan, as I needed exactly something like this. I might consider trying a port myself, but I'm still a total newbie in Haxe world, so I'm not sure how long that will take and how well I can do it.

Thanks!

**Comment by peteshand** [comment]
Hi @ValentinRadoi 
We currently don't have this on a milestone, so it's a little hard to tell you exactly when this will get implemented, however hopefully as a community we will add the new AssetManager over the coming weeks/months.

**Comment by ValentinRadoi** [comment]
Hi @peteshand 
Thanks for the quick response!

In that case I will attempt at implementing it myself, I guess. For now I've forked this repo (https://github.com/ValentinRadoi/starling) and will do some upgrades from here https://github.com/Gamua/Starling-Framework/commit/53ff6244cf2a9349f1eb5246acb36868feec5bab onwards, towards the present, going in parallel with original Starling development, until I get to the new AssetsManager.

Can't promise anything, but if I have something worth showing, I will do a pull request.

**Comment by jgranick** [comment]
How I update is use a good merge tool (such as "WinMerge" or "Meld") and do a comparison between the last AS3 sources we used (https://github.com/gamua/Starling-Framework/commit/ceb4ecad266dad4a076d387b35d0d36ecd03f40e) and the latest coming from the official. Sometimes, it's easier to look only at that comparison, and copy in the changes by hand.

For larger merges, I take the latest sources, and run a file rename (*.as to *.hx) then I edit files and decrease the indent by one (Ctrl + A, Shift + Tab, save).

Then I can run merge on the _slightly_ modified AS3 sources against our current Haxe source, this gives a good comparison from our Haxe sources to the AS3 sources, and obviously there are some differences, but a lot of the code and formatting is still identical. This really helps if we have a bug, so we can compare to the original and see if we ported something incorrectly
